File: logisticRegression.py
# -*- coding: utf-8 -*-
"""
Created on Sun Dec 29 11:47:53 2019

@author: 138709
"""

# Logistic Regression


# Import the libraries
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt


# Import the dataset
dataset = pd.read_csv('Social_Network_Ads.csv')
X = dataset.iloc[:,[2,3]].values
Y = dataset.iloc[:,4].values


# Plotting to see the relation between the variables
plt.scatter(dataset.iloc[:,2].values, dataset.iloc[:,4].values)
plt.title('Social Network')
plt.xlabel('Age')
plt.ylabel('Purchased')
plt.show()

plt.scatter(dataset.iloc[:,3].values, dataset.iloc[:,4].values)
plt.title('Social Network')
plt.xlabel('Estimates Salary')
plt.ylabel('Purchased')
plt.show()


# Split the dataset into train and test set
from sklearn.cross_validation import train_test_split
X_train, X_test, Y_train, Y_test = train_test_split(X, Y, test_size = 0.25, random_state = 0)


# Feature Scaling the dataset
from sklearn.preprocessing import StandardScaler
scaler = StandardScaler()
scaler.fit(X_train)
X_train = scaler.transform(X_train)
X_test = scaler.transform(X_test)


# Fitting Logistic Regression
from sklearn.linear_model import LogisticRegression
model = LogisticRegression(random_state = 0)
model.fit(X_train, Y_train)


# Predicting the test set results
Y_pred = model.predict(X_test)


# Making the confusion matrix
from sklearn.metrics import confusion_matrix
results = confusion_matrix(Y_test, Y_pred)
print(results)

File: logisticRegression.R
install.packages("ggplot2")
library(ggplot2)
install.packages("caTools")
library(caTools)
install.packages("caret")
library(caret)

# Logistic Regression


# Import the dataset
dataset <- read.csv('Social_Network_Ads.csv')
dataset <- dataset[,3:5]


# Plotting to see the relation between points
ggplot(dataset, aes(dataset[,1], dataset[,3])) + geom_point() + ggtitle('Visualisation') + xlab('Age') + ylab('Purchased')
ggplot(dataset, aes(dataset[,2], dataset[,3])) + geom_point() + ggtitle('Visualisation') + xlab('Estimated Salary') + ylab('Purchased')


# Splitting the dataset into train and test sets
set.seed(123)
sample = sample.split(dataset, SplitRatio = 3/4)
train = subset(dataset, sample == TRUE)
test = subset(dataset, sample == FALSE)


# Standard Scaling in R
train[,1:2] = scale(train[,1:2])
test[,1:2] = scale(test[,1:2])


# Fitting Logistic Regression to our dataset
model = glm(formula = Purchased ~ Age + EstimatedSalary, family = binomial, data = train)


# Predicting the results
prediction = predict(model, newdata = test[,-3], type = "response")
predicted = ifelse(prediction > 0.5, 1, 0)


# Making the confusion matrix
table(test[,3], predicted)